vms/evm/customlogs/log_ext.go

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package customlogs
+
+import "github.com/ava-labs/libevm/core/types"
+
+// FlattenLogs converts a nested array of logs to a single array of logs.
+func FlattenLogs(list [][]*types.Log) []*types.Log {
```

Comment:
I think the usage of this in `core` makes sense... but in `filters` we just use this to iterate over the logs and check if they match a filter ([ref](https://github.com/ava-labs/subnet-evm/blob/53f53056e9d161ada1ea42506b9537755018715d/eth/filters/filter.go#L406)). We don't need to flatten the logs to do this... we could just iterate in 2-D. I feel like we could move this to `core/blockchain.go` pre-emptively instead and have it live as an unexported helper in that package where it is needed, and just update `filters` to just perform a 2-d iteration over the logs.